contracts/transfer/src/state.rs

```diff
@@ -555,6 +590,37 @@ impl TransferState {
             .then(|| ())
             .ok_or(Error::ProofVerificationError)
     }
+
+    pub fn tx_hash(
```

Comment:
Code for hashing shouldn't really be mixed in to the contract. The code size is very large and is quite expensive. The host methods included by rusk should serve this purpose just fine.
Take a look at the new `rusk-abi` tests.

Reply:
This is also already handled in `transfer_types::Transaction::hash`.

Reply:
done, initially I thought there is no default constructor for Proof, that was the reason I re-introduced static tx_hash